frontend/ui/translator.lua

```diff
@@ -377,6 +387,20 @@ function Translator:loadPage(text, target_lang, source_lang)
     local query = ""
     self.trans_params.tl = target_lang
     self.trans_params.sl = source_lang
+    if G_reader_settings:isTrue("translator_from_romanizations") then
+       table.insert(self.trans_params.dt, "rm")
+    else
+       local rmIndex = nil
+       for i, v in pairs(self.trans_params.dt) do
+          if v == "rm" then
+             rmIndex = i
```

Comment:
No other way avoid this and the ugly removal? Dunno (no time to check) how this appears in the query when dt is an array.

Reply:
Hmm, you're right. 

Actually, I only had to do this to make the tests that I added pass since it reuses the `Translator` between tests. Actually, I'm pretty sure from testing it for real that this doesn't need to be done otherwise. 

Do you have a suggestion how to better handle using this in tests?

Reply:
I moved the initialization of `dt` to this function so that it doesn't affect the tests. Now the tests still pass and it's not as ugly. Thanks for asking for me to look at this again.